asm/load_store.go

```diff
@@ -26,10 +28,119 @@ const (
 	MemMode Mode = 0x60
 	// MemSXMode - load from memory, sign extension
 	MemSXMode Mode = 0x80
-	// XAddMode - add atomically across processors.
-	XAddMode Mode = 0xc0
+	// AtomicMode - add atomically across processors.
+	AtomicMode Mode = 0xc0
 )
 
+const atomicMask OpCode = 0x0001_ff00
+
+type AtomicOp uint32
+
+const (
+	InvalidAtomic AtomicOp = 0xffff_ffff
+
+	// AddAtomic - add src to memory address dst atomically
+	AddAtomic AtomicOp = AtomicOp(Add) << 8
+	// FetchAdd - add src to memory address dst atomically, store result in src
+	FetchAdd AtomicOp = AddAtomic | fetch
```

Comment:
@dylandreimerink I exported these. Then you can just write `asm.FetchAdd.Mem(...)` which is shorter than `asm.AddAtomic.Fetch(...)`. I suspect the Fetch variants might be more commonly used anyways. What do you think?

Reply:
Yes, this seems nice!